### PR TITLE
Add functions to post to social media

### PR DESCRIPTION
**Pull request by archangelic**
This should resolve issue #10 and have mastodon functionality to boot!

### CODE REVIEW

File: ttadmin/common/social.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    client_secret=settings.MASTO_CLIENT_SECRET,
+    access_token=settings.MASTO_ACCESS_TOKEN,
+    api_base_url=settings.MASTO_BASE_URL
+)
+
+tw_auth = tweepy.OAuthHandler(settings.TWITTER_CONSUMER_KEY, settings.TWITTER_CONSUMER_SECRET)
+tw_auth.set_acces_token(settings.TWITTER_TOKEN, settings.TWITTER_TOKEN_SECRET)
+twitter = tweepy.API(tw_auth)
+
+def post_to_social(qs):
```
Comment (vilmibm) [inline ttadmin/common/social.py]: looking good! thank you!!!

couple thoughts:

- it'd be nice to have two separate functions: `post_to_mastodon` and `post_to_twitter` that `post_to_social` can then use
- what if the list of new users exceeds the limit for twitter?
- i guess the same question for mastodon but it seems very very unlikely

File: ttadmin/common/social.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    users = ''
+    for townie in qs:
+        users += '~{}\n'.format(townie.username)
+    users = users.strip()
+    if len(qs) != 0:
```
Comment (vilmibm) [inline ttadmin/common/social.py]: shouldn't this be `len(qs) > 1`?

**Review by vilmibm — CHANGES_REQUESTED** [review]

File: ttadmin/common/social.py (hunk 1 above)
Comment (archangelic) [reply ttadmin/common/social.py]: I guess we should post into two different posts, the question is, should the welcome part be identical?

Also, should `post_to_social` just pass the queryset on to the other functions?

File: ttadmin/common/social.py (hunk 2 above)
Comment (archangelic) [reply ttadmin/common/social.py]: Wow, that was silly. thanks for catching that!

**Comment by archangelic** [comment]
Okay, made some significant changes to the code, let me know how that looks!

File: ttadmin/common/social.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    access_token=settings.MASTO_ACCESS_TOKEN,
+    api_base_url=settings.MASTO_BASE_URL
+)
+
+tw_auth = tweepy.OAuthHandler(settings.TWITTER_CONSUMER_KEY, settings.TWITTER_CONSUMER_SECRET)
+tw_auth.set_access_token(settings.TWITTER_TOKEN, settings.TWITTER_TOKEN_SECRET)
+twitter = tweepy.API(tw_auth)
+
+def split_posts_by_length(text, length):
+    pattern = '.{,%d}(?:\s|$)' % length - 1
```
Comment (vilmibm) [inline ttadmin/common/social.py:20]: clever!

**Review by vilmibm — APPROVED** [review]
hell yeah, thanks so much!

**Pull request merged**